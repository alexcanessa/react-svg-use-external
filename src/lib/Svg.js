/* @flow */

import * as React from "react";

type Props = {
  children?: React.Node,
  forwardedRef?: React.Ref<"svg">,
  [key: string]: any
};

type State = {
  viewBox: ?string
};

export const Context = React.createContext();

class Svg extends React.Component<Props, State> {
  state = {
    viewBox: undefined
  };

  _setViewBox(viewBox) {
    this.setState({ viewBox });
  }

  _context = {
    setViewBox: this._setViewBox.bind(this)
  };

  _href = null;

  render() {
    const { children, viewBox, forwardedRef, ...props } = this.props;
    return (
      <svg
        viewBox={viewBox || this.state.viewBox}
        ref={forwardedRef}
        {...props}
      >
        <Context.Provider value={this._context}>{children}</Context.Provider>
      </svg>
    );
  }
}

export default React.forwardRef((props, ref) => (
  <Svg {...props} forwardedRef={ref} />
));
