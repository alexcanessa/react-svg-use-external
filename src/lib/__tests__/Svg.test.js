import React from "react";
import Svg from "../Svg";
import { render } from "react-testing-library";

describe("Svg", () => {
  test("should render an <svg> element", () => {
    const { container } = render(<Svg />);
    expect(container.firstChild).toMatchSnapshot();
  });
  test("should pass through attributes and children", () => {
    const { container } = render(
      <Svg data-foo="bar" viewBox="0 0 1 1">
        <g />
        <circle />
      </Svg>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
  test("should expose an <svg> ref", () => {
    const ref = React.createRef();

    render(<Svg ref={ref} />);
    expect(ref.current).toBeInstanceOf(Element);
    expect(ref.current.tagName).toMatch(/^svg$/i);
  });
});
