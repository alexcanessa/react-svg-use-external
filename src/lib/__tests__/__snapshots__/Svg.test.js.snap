// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg should pass through attributes and children 1`] = `
<svg
  viewBox="0 0 1 1"
  data-foo="bar"
>
  <g />
  <circle />
</svg>
`;

exports[`Svg should render an <svg> element 1`] = `<svg />`;
